gateway/1.5/modules/ROOT/pages/policies-pdk-configure-features.adoc

```diff
@@ -1,5 +1,589 @@
+}
+
+async fn response_filter(state: ResponseState, RequestData(path): RequestData<String>) {
+    let state = state.into_headers_state().await;
+    logger::info!("Path: {path}, Status: {}", state.status_code());
+}
+
+#[entrypoint]
+async fn configure(launcher: Launcher) -> Result<()> {
+    let filter = on_request(request_filter)
+        .on_response(response_filter);
+
+    launcher
+        .launch(filter)
+        .await
+}
+----
+
+Returning a `RequestData` makes it impossible for a response to reach a client early. This is why the `Flow::Continue` enum has a template parameter. Implement this by adding a `on_response` wrapped function to the `Flow` example used in <<stop-request-execution>>:
+
+[source,Rust]
+----
+async fn request_filter(state: RequestState) -> Flow<String> {
+    let state = state.into_headers_state().await;
+    if state.header("authorization").is_some() {
+        Flow::Continue(state.path())
+    } else {
+        Flow::Break(Response::new(401)
+            .with_headers(vec![("WWW-Authenticate".to_string(), "Bearer realm=\"oauth2\"".to_string())])
+            .with_body(r#"{ "error": "token was not present"}"#))
+    }
+}
+
+async fn response_filter(state: ResponseState, RequestData(path): RequestData<String>) {
+    let state = state.into_headers_state().await;
+    logger::info!("Path: {path}, Status: {}", state.status_code());
+}
+----
+
+== Share Data Between Workers and Caching
+
+To increase the capability of handling requests, envoy spawns many workers for the policy. A single worker handles all the stages of a specific request including outgoing connections and the response flow. Due to this, you can track an internal status for a specific request and create sequential execution flow even though the underlying implementation is event driven. 
+
+Each worker processes a single request at a given time. To share a global state in a specific worker, use a `RefCell` without any risk of concurrent modification.
+
+Communicating a shared global state to all workers is more complex. Achieve this is by using the provided `Cache` mechanism.
+
+After injecting the cache builder, provide an ID for the cache and the maximum number of elements the cache can support. 
+
+For example a cache with id “calculator” can hold ten elements:
+
+[source,Rust]
+----
+#[entrypoint]
+async fn configure(
+    launcher: Launcher,
+    cache_builder: CacheBuilder,
+) -> Result<()> {
+    let cache = cache_builder.build(String::from("calculator"), 10);
+    ...
+----
+
+The cache has the following interface:
+
+[source,Rust]
+----
+pub trait Cache {
+    fn save(&self, key: &str, value: Vec<u8>) -> Result<(), CacheError>;
+    fn get(&self, key: &str) -> Option<Vec<u8>>;
+    fn delete(&self, key: &str) -> Option<Vec<u8>>;
+    fn purge(&self);
+}
+----
+
+IMPORTANT: Because another work may write to the cache concurrently, there is no guarantee that the value overwritten when saving the cache is the same as when it was retrieved.
```

Comment:
```suggestion
IMPORTANT: Because different works write to the cache concurrently, there is no guarantee that the value overwritten when saving the cache is the same as when it was retrieved.
```